src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -26,6 +28,7 @@ public class HttpRequestMessage : IDisposable
         private HttpContent? _content;
         private bool _disposed;
         private IDictionary<string, object?>? _properties;
+        private HttpRequestMessageFinalizer? _finalizer;
```

Comment:
Is there another place we can track this? Maybe as a bit flag in `_sendStatus`? It would be nice to not increase the size of `HttpRequestMessage`.

Reply:
bonus points: merge `_disposed` into `_sendStatus` too.

Reply:
Yes, please don't increase the size of the object for this.  We should also avoid adding a finalizer.  Even if finalization is suppressed, it makes object creation measurably more expensive.

Reply:
I believe we could avoid it by instead using Http(2)Connection's finalizer - it would mean having to remove [this optimization](https://github.com/dotnet/runtime/blob/87919a0462bc2ebd5c256e5ad7649ff98c858e2c/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1158-L1163) when Telemetry is enabled + adding a finalizer to Http2Connection if Telemetry is enabled.

Reply:
I'm not familiar with this code, but I will add +1 to @stephentoub's comment.  We should not add a finalizable object for this.  It's definitely much more expensive, always gets promoted, and if you only do it when tracing is on, then it more noticeably changes the behavior with tracing on.  I also worry about adding too much "just in tracing" logic, since it often magnifies the observer effect, meaning that behavior changes when you're watching with the profiler attached.

In general, we just accept when we get missing start or stop events and make tools that know how to handle them, but with counters, that changes things a bit, since you have a running total that you're trying to track vs. just missing a start or stop for a pair of events.  But do remember that it's still possible to be missing starts and stops even if you do everything perfectly, because there's always a race between emitting events and enabling or disabling tracing.